src/dos/dos.cpp

```diff
@@ -700,29 +708,32 @@ static Bitu DOS_21Handler(void) {
 				LOG(LOG_DOSMISC,LOG_ERROR)("Someone playing with cpsw %x",reg_ax);
 				break;
 			case 5:reg_dl=3;break;//TODO should be z						/* Always boot from c: :) */
-			case 6:											/* Get true version number */
-				reg_bl=dos.version.major;
-				reg_bh=dos.version.minor;
-				reg_dl=dos.version.revision;
-				reg_dh=0x10;								/* Dos in HMA */
-				break;
-			default:
-				LOG(LOG_DOSMISC,LOG_ERROR)("Weird 0x33 call %2X",reg_al);
-				reg_al =0xff;
-				break;
-		}
-		break;
-	case 0x34:		/* Get INDos Flag */
-		SegSet16(es,DOS_SDA_SEG);
-		reg_bx=DOS_SDA_OFS + 0x01;
-		break;
-	case 0x35:		/* Get interrupt vector */
-		reg_bx=real_readw(0,((uint16_t)reg_al)*4);
-		SegSet16(es,real_readw(0,((uint16_t)reg_al)*4+2));
-		break;
-	case 0x36:		/* Get Free Disk Space */
-		{
-			uint16_t bytes,clusters,free;
+		        case 6: // Get true version number, not affected by SETVER
+		        {
+			        DOS_PSP psp(dos.psp());
+			        reg_bl = dos.version.major;
+			        reg_bh = dos.version.minor;
+			        reg_dl = dos.version.revision;
+			        reg_dh = 0x10; // DOS in HMA
+		        } break;
+		        default:
+			        LOG(LOG_DOSMISC, LOG_ERROR)
+			        ("Weird 0x33 call %2X", reg_al);
+			        reg_al = 0xff;
+			        break;
+		        }
+		        break;
+	case 0x34: /* Get INDos Flag */
+		        SegSet16(es, DOS_SDA_SEG);
+		        reg_bx = DOS_SDA_OFS + 0x01;
+		        break;
+	case 0x35: /* Get interrupt vector */
+		        reg_bx = real_readw(0, ((uint16_t)reg_al) * 4);
+		        SegSet16(es, real_readw(0, ((uint16_t)reg_al) * 4 + 2));
```

Comment:
I'm having trouble seeing context here - is the indenting on line 725 and prior due to an inner switch statement inside of an outer case?

(Just mentioning it - sometimes the clang formatter can incorrectly format switches)

Reply:
I have reformatted the code manually, in a separate commit for better bisectability.